contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -732,6 +732,48 @@ static VALUE execute_read_query(VALUE vargs)
     return result;
 }
 
+static VALUE rb_next_result(VALUE self)
+{
+    struct trilogy_ctx *ctx = get_open_ctx(self);
+
+    // TODO: Should this return nil or raise an error?
```

Comment:
Actually, a follow up question: I'm assuming `ctx->conn.server_status & TRILOGY_SERVER_STATUS_MORE_RESULTS_EXISTS` means "the server status is non-zero AND we have more results available". And if I'm understanding correctly, the situation in this conditional is the one where we've called `#next_result` without more results existing. 

Should this conditional be `if ctx->conn.server_status & !TRILOGY_SERVER_STATUS_MORE_RESULTS_EXISTS` instead, ie. "if the server status is non-zero AND we have no more results, then return nil"?

Reply:
You're right in that what we're after is returning `nil` if `#next_result` is called despite the server not having any more results.

I think perhaps you've misinterpreted the bitwise AND (&) operator for a boolean AND operation instead. 

In this case, `ctx->conn.server_status` is an unsigned 16-bit integer which can be encoded with a series of `SERVER_STATUS` integers through bitwise OR'ing (|) them into the status.

`ctx->conn.server_status & TRILOGY_SERVER_STATUS_MORE_RESULTS_EXISTS`

This conditional is effectively checking if `TRILOGY_SERVER_STATUS_MORE_RESULTS_EXISTS` (0x0008) is present in the server_status because it will only return a non-zero result if those bits are flipped on in the status.

Then, we negate this to effectively check if it **wasn't present**, which is what we're after.

Reply:
Ah, yep that makes sense! I indeed misread that line 🤦‍♀️  I didn't realize that `ctx->conn.server_status` was the result of bitwise OR'ing `SERVER_STATUS` flags, and I was thrown off by what `ctx->conn.server_status` was representing. 

Thanks for elaborating! ❤️